internal/c/libqb.cpp

```diff
@@ -35923,6 +35923,48 @@ void sub__echo(qbs *message) {
 
 } // echo
 
+qbs *func__readfile(qbs *filespec)
+{
+    FILE *file; int len;   // file handle; file length;
+    qbs *namez; qbs *cont; // 0-term file name; file contents;
+
+    namez = qbs_add(filespec, func_chr(0)); // add terminator
+    filepath_fix_directory(namez);          // fix separators
+    file = fopen((const char*)namez -> chr, "rb");
+
+    if (file) {
+        fseek(file, 0, SEEK_END); // end pos
+        len = ftell(file);        //   = file length
+        rewind(file);             // rewind to start
+        cont = qbs_new(len, 1);   // get new string for file contents
+        fread(cont -> chr, 1, len, file);
+        fclose(file);
+        return cont;
+    } else {
+        error(QB_ERROR_FILE_NOT_FOUND); // error, we throw most common
+        return qbs_new_txt(""); // return empty on error
+    }
+}
+
+int16_t func__writefile(qbs *filespec, qbs *contents)

```

Comment:
Shouldn't this be a `SUB` instead of a `FUNCTION`?

We are already throwing an error. So, we can expect folks to trap the error if writing fails. Basically, either return a failure code or throw an error, not both.

I would side with throwing an error because it makes the usage simpler.